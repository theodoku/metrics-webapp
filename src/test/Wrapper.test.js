import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import { render } from '@testing-library/react';
import Wrapper from '../components/wrapper.js';

describe('Wrapper component', () => {
  const wrapper = {
    name: 'Test Wrapper',
    population: '1000000',
    map: 'test-map.png',
  };

  it('should match snapshot', () => {
    const { container } = render(<Wrapper wrapper={wrapper} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the name and population', () => {
    const { getByText } = render(<Wrapper wrapper={wrapper} />);
    expect(getByText(wrapper.name)).toBeInTheDocument();
    expect(getByText(wrapper.population)).toBeInTheDocument();
  });

  it('should render the map image with the correct alt text', () => {
    const { getByAltText } = render(<Wrapper wrapper={wrapper} />);
    expect(getByAltText(wrapper.name)).toBeInTheDocument();
  });
});
