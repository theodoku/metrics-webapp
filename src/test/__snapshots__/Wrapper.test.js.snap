// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper component should match snapshot 1`] = `
<div
  class="right-arrow"
>
  <svg
    aria-hidden="true"
    fill="none"
    height="1em"
    stroke="currentColor"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M13 9l3 3m0 0l-3 3m3-3H8m13 0a9 9 0 11-18 0 9 9 0 0118 0z"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
</div>
`;
